src/main/webapp/mapsScript.js

```diff
@@ -0,0 +1,79 @@
+ /*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Fetch apiKey and dynamically load the Maps JavaScript API.
+ */
+function loadMapsScript() {
+  fetch('apiKey.json').then(response => response.json()).then(jsonResponse => {
+    GoogleMapsApiKey = jsonResponse.apiKey;
+    // Create the Google Maps tag with the fetched api key.
+    let script = document.createElement('script');
+    script.src = `https://maps.googleapis.com/maps/api/js?key=${GoogleMapsApiKey}`;
+    // Append the 'script' element to 'head'.
+    document.body.appendChild(script);
+  });
+}
+
+/**
+ * Creates and returns a Map centered at (centerLat, centerLng). 
+ */
+function initMap(centerLat, centerLng) {
+  const map = new google.maps.Map(
+      document.getElementById('map'),
+      {center: {lat: centerLat, lng: centerLng}, zoom: 12});
+  return map;
```

Comment:
Might be worth allowing for an optional zoom level? You can provide a default value here if the caller does not provide one.

```suggestion
function initMap(centerLat, centerLng, zoomLevel = 12) {
  const map = new google.maps.Map(
      document.getElementById('map'),
      {center: {lat: centerLat, lng: centerLng}, zoom: zoomLevel});
  return map;
```

Reply:
Yes, I added it.